File: src/data/courses/dsa/two-pointers-sliding-window/two-pointers-opposite-direction/sort-colors/code/cpp/sort_colors_sol1.cpp
class Solution {
public:
    // Counting Sort: Two passes
    void sortColors(vector<int>& nums) {
        int count[3] = {};
        for (int n : nums) count[n]++;         // Pass 1: Count
        int idx = 0;
        for (int c = 0; c < 3; c++)            // Pass 2: Write
            for (int i = 0; i < count[c]; i++)
                nums[idx++] = c;
    }
};

File: src/data/courses/dsa/two-pointers-sliding-window/two-pointers-opposite-direction/sort-colors/code/java/SortColorsSol1.java
class Solution {
    public void sortColors(int[] nums) {
        int[] cnt = new int[3];
        for (int n : nums) cnt[n]++;
        int idx = 0;
        for (int c = 0; c < 3; c++) for (int i = 0; i < cnt[c]; i++) nums[idx++] = c;
    }
}